### PR TITLE
Readme for introsort

### PR DESCRIPTION
**Pull request by raksha009**
Fixes #2249 

#### Checklist:
- [ ] 4 space indentation.
- [ ] Coding conventions are followed.
- [ ] Input is taken dynamically.
- [ ] Sample Input / Output is added at the end of file.
- [X] Logic Documentation (Comments).
- [X] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Readme added
- 

#### Languages Used:
- 
- 

#### Files Added:
- README.md
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Intro_Sort/README.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+1. It was invented by David Musser in 1997.
+
+2. It is also known as introspective sort.
+
+3. It is a hybrid sorting algorithm(combines two or more other algorithms that solve the same problem,either choosing one depending upon the data or switching between them over the course of algorithm).
+
+4. It provides both fast average performance and optimal worst performance,thus allowing the performance requirement to become more strict and effective.
+
+5. It combines the good part of three algorithm i.e quick sort, heap sort,insertion sort.
```
Comment (asha15) [inline Intro_Sort/README.md:11]: the details is not enough. please include some more. if possible please add a GIF

File: Intro_Sort/README.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+
```
Comment (asha15) [inline Intro_Sort/README.md:13]: remove the extra lines

File: Intro_Sort/README.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+3. It is a hybrid sorting algorithm(combines two or more other algorithms that solve the same problem,either choosing one depending upon the data or switching between them over the course of algorithm).
+
+4. It provides both fast average performance and optimal worst performance,thus allowing the performance requirement to become more strict and effective.
+
+5. It combines the good part of three algorithm i.e quick sort, heap sort,insertion sort.
+
+   
+
+   ## Pseudocode
```
Comment (asha15) [inline Intro_Sort/README.md:15]: remove the space at the beginning

File: Intro_Sort/README.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+2. It is also known as introspective sort.
+
+3. It is a hybrid sorting algorithm(combines two or more other algorithms that solve the same problem,either choosing one depending upon the data or switching between them over the course of algorithm).
+
+4. It provides both fast average performance and optimal worst performance,thus allowing the performance requirement to become more strict and effective.
+
+5. It combines the good part of three algorithm i.e quick sort, heap sort,insertion sort.
+
+   
+
```
Comment (asha15) [inline Intro_Sort/README.md:14]: please include the Algorithm

File: Intro_Sort/README.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+
+**Step 2:** From the previous step, begin=0, end=29, a[pivot]=8 The recursion starts for the sub-array a[begin,pivot-1] and a[pivot+1,end] 2,2,4,2,4,8,0,3,8, 8 ,28,16,9,11,28,10,13,24,22,28,27,13,27,13,27,13,24,23,18,22,10,10 depthLimit = 7, depthLimit = 7, begin = 0,end = 8 begin = 10,end = 29 number of element <= 16 Number of element > 16,hence sortDataUtil(10,29) that calculates the pivot and a[partition]=27. 
+
+Hence insertionSort (0,8)
+
+0,2,2,2,3,4,4,8,8, 8, 16,9,11,10,13,24,22,10,13,27,13,24,23,18,22,10, 27, 28,28,28
+
+
```
Comment (asha15) [inline Intro_Sort/README.md:52]: the example looks more compact. please keep it simple and clear

**Comment by raksha009** [comment]
@asha15  Maam, i have made the changes. Also, added the image from which i understood and hence took that as an example.

File: Intro_Sort/README.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+**Time Complexity**
+
+Average performance O(n log n) Optimal worst performance O(n log n) where, n = number of elements to be sorted.
+
+SEE ALSO:
+
+https://github.com/jainaman224/Algo_Ds_Notes/blob/master/Intro_Sort/introsort.py
```
Comment (singh-shreya6) [inline Intro_Sort/README.md:108]: Please add References. You must have referred to articles and add source for images. A reference section is missing here.

File: Intro_Sort/README.md [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+
+
+
+**Step 4:** 
+
+output: 0,2,2,2,3,4,4,8,8,8,9,10,10,11,13,13,13,16,18,22,22,23,24,24,27,27,28,28,28 
+
+In the above example, the depthLimit came until 6, and for bigger datasets the depthLimit might reach 0 for certain sub-array. The heapSort is carried out only for those sub-array for sorting them.
+
+Output: 0,2,2,2,3,4,4,8,8,8,9,10,10,10,11 13,13,13,16,18,22,22,23,24,24,27,27,28 28,28
```
Comment (singh-shreya6) [inline Intro_Sort/README.md:98]: Please add space after comma

File: Intro_Sort/README.md [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+
+​                         `Sorted!                               Sorted!                    Sorted!                         `
+
+Hence, no further recursion hence no further recursion hence no further recursion.
+
+
+
+**Step 4:** 
+
+output: 0,2,2,2,3,4,4,8,8,8,9,10,10,11,13,13,13,16,18,22,22,23,24,24,27,27,28,28,28
```
Comment (singh-shreya6) [inline Intro_Sort/README.md:94]: Please add space after comma.

File: Intro_Sort/README.md [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+
+The recursion starts for the sub-array a[begin,partition-1] and a[partition+1,end]
+
+` 0,2,2,2,3,4,4,8,8,8,16,9,11,10,13,24,22,10,13,27,13,24,23,18,22,10,27,28,28,28     ` 
+
+ `sorted!                         depthLimit=6                             depthLimit=6 `
+
+Hence,no further recursion.
+
+ 	begin = 10, 	end = 25 	begin = 10, 	end = 25
```
Comment (singh-shreya6) [inline Intro_Sort/README.md:78]: This looks different in actual view of readme. Please fix this line.

File: Intro_Sort/README.md [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+
+
+
+**Step 3:** From the previous step, 
+
+depthLimit = 6,	begin = 10, 	end = 29 
+
+The recursion starts for the sub-array a[begin,partition-1] and a[partition+1,end]
+
+` 0,2,2,2,3,4,4,8,8,8,16,9,11,10,13,24,22,10,13,27,13,24,23,18,22,10,27,28,28,28     `
```
Comment (singh-shreya6) [inline Intro_Sort/README.md:72]: Space after comma and break in into 2 lines.

File: Intro_Sort/README.md [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+
+0,2,2,2,3,4,4,8,8, 8, 16,9,11,10,13,24,22,10,13,27,13,24,23,18,22,10, 27, 28,28,28
+
+
+
+**Step 3:** From the previous step, 
+
+depthLimit = 6,	begin = 10, 	end = 29 
+
+The recursion starts for the sub-array a[begin,partition-1] and a[partition+1,end]
```
Comment (singh-shreya6) [inline Intro_Sort/README.md:70]: Space around - and +

File: Intro_Sort/README.md [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+
+depthLimit = 7, 	depthLimit = 7, 	begin = 0,	end = 8 	begin = 10,	end = 29 
+
+number of element <= 16 Number of element > 16,
+
+hence sortDataUtil(10,29) that calculates the pivot and a[partition]=27. 
+
+Hence insertionSort (0,8)
+
+0,2,2,2,3,4,4,8,8, 8, 16,9,11,10,13,24,22,10,13,27,13,24,23,18,22,10, 27, 28,28,28
```
Comment (singh-shreya6) [inline Intro_Sort/README.md:62]: Inconsistency in space. Keep it after all commas

File: Intro_Sort/README.md [hunk 13]
```diff
@@ -0,0 +1,108 @@
...
+
+he recursion starts for the sub-array a[begin,pivot-1] and a[pivot+1,end] 
+
+2,2,4,2,4,8,0,3,8, 8 ,28,16,9,11,28,10,13,24,22,28,27,13,27,13,27,13,24,23,18,22,10,10 
+
+depthLimit = 7, 	depthLimit = 7, 	begin = 0,	end = 8 	begin = 10,	end = 29 
+
+number of element <= 16 Number of element > 16,
+
+hence sortDataUtil(10,29) that calculates the pivot and a[partition]=27.
```
Comment (singh-shreya6) [inline Intro_Sort/README.md:58]: Space around =, also the first character of new line should begin with Capital letters.

**Comment by raksha009** [comment]
Done the changes plz review and merge my pr @singh-shreya6 @asha15 maam

File: Intro_Sort/README.md [hunk 14]
```diff
@@ -0,0 +1,115 @@
...
+
+Number of element <= 16 Number of element > 16,
+
+Hence sortDataUtil(10,29) that calculates the pivot and a[partition] = 27. 
+
+Hence insertionSort (0,8)
+
+0, 2, 2, 2, 3, 4, 4, 8, 8, 8, 16, 9, 11, 10, 13, 24, 22, 10, 13,  27, 13, 24, 23, 18, 22, 10, 27, 28, 28, 28
+
+
```
Comment (asha15) [inline Intro_Sort/README.md:66]: remove these lines

File: Intro_Sort/README.md [hunk 15]
```diff
@@ -0,0 +1,115 @@
...
+**Step 4:** 
+
+Output: 0, 2, 2, 2, 3, 4, 4, 8, 8, 8, 9, 10, 10, 11, 13, 13, 13, 16, 18, 22, 22, 23, 24, 24, 27, 27, 28, 28, 28 
+
+In the above example, the depthLimit came until 6, and for bigger datasets the depthLimit might reach 0 for certain sub-array. The heapSort is carried out only for those sub-array for sorting them.
+
+Output: 0, 2, 2, 2, 3, 4, 4, 8, 8, 8, 9, 10, 10, 10, 11, 13, 13, 13, 16, 18, 22, 22, 23, 24, 24, 27, 27, 28, 28, 28
+
+
```
Comment (asha15) [inline Intro_Sort/README.md:102]: remove these lines

File: Intro_Sort/README.md [hunk 16]
```diff
@@ -0,0 +1,115 @@
...
+
+**Time Complexity**
```
Comment (asha15) [inline Intro_Sort/README.md:104]: make this as a sub header

File: Intro_Sort/README.md [hunk 17]
```diff
@@ -0,0 +1,115 @@
...
+
+Average performance O(n log n) Optimal worst performance O(n log n) where, n = number of elements to be sorted.
+
+**SEE ALSO :**
```
Comment (asha15) [inline Intro_Sort/README.md:108]: make this as a sub header

File: Intro_Sort/README.md [hunk 18]
```diff
@@ -0,0 +1,115 @@
...
+
+https://github.com/jainaman224/Algo_Ds_Notes/blob/master/Intro_Sort/introsort.py
+
+**REFERENCES** :
```
Comment (asha15) [inline Intro_Sort/README.md:112]: make this as a sub header

**Comment by raksha009** [comment]
Done the changes @asha15 maam, but i  couldn't make reduce lines in readme file on my desktop as its not accepting it.  So, made the changes here itself. Hence I won't be able to squash the commits.
Sorry for inconvenience.

**Pull request merged**